lib.php

```diff
@@ -1255,6 +1282,37 @@ function urkund_get_form_elements($mform) {
                        $supportedfiles, array('multiple' => true));
     $mform->setType('urkund_selectfiletypes', PARAM_TAGLIST);
 
+    // Check if maharaws is enabled.
+    $pm = \core_plugin_manager::instance();
+    $submissionplugins = $pm->get_enabled_plugins('assignsubmission');
+    if (!empty($submissionplugins) && key_exists('maharaws', $submissionplugins)) {
+        $mform->addElement('html', get_string('setmahara', 'plagiarism_urkund'));
+
+        $mform->addElement('text', 'urkund_maharawsurl', get_string('maharawsurl', 'plagiarism_urkund'),
```

Comment:
storing this stuff at the activity level and allowing the teacher to configure feels very wrong - we should not be exposing secrets/keys to normal teaching staff.

The keys/urls secrets should be just stored at the admin level - not within the course module editing page. I suspect Murdoch only have a single mahara instance in use so storing this in one place shouldn't be a problem for them.

If we need to support multiple Mahara instances from the same Moodle site, these should still be set up at the site level, and then a drop list shown on the course module editing page allowing the teacher to select which Mahara site they are using, pulling the keys/urls from site-level settings.

Reply:
For initial release, I'd reccomend removing these from the teachers settings completly and just give them the abilty to turn "get mahara stuff" on/off as a single setting.